.github/actions/upload-log/action.yml

```diff
@@ -0,0 +1,25 @@
+name: Upload the log file while running
+
+description: Upload the log file while the job runs
+
+inputs:
+  job-id:
+    description: Job id
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Upload log file in background
```

Comment:
This looks ok IMO, but I think we should gate this only to EC2 instances.  Uploading to S3 from EC2 instances of the same region is practically free.  As we are not 100% sure about the benefit of doing this, I kind of want to exclude GCP A100 runner.  It's also use `_linux-build` and `_linux_test` workflows, i.e. https://github.com/pytorch/pytorch/blob/master/.github/workflows/inductor.yml#L56